isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/serviceschicht.adoc

```diff
@@ -0,0 +1,119 @@
+Die Umwandlung kann programmatisch oder über das Produkt https://mapstruct.org/[MapStruct] erfolgen.
+
+Fehlerbehandlung:: Fachliche Fehler, die bei der Verarbeitung einer Anfrage in der Serviceschicht oder im Anwendungskern auftreten, müssen von der Fachkomponente an den Aufrufer zurückgegeben werden.
+Hierzu verwenden Backends den xref:isyfact-standards-doku:isy-exception-core:konzept/master.adoc[Baustein Fehlerbehandlung].
+
+Im Normalfall ist der Funktionsumfang der Fachkomponenten der Serviceschicht viel geringer als der Funktionsumfang ihrer Entsprechungen im Anwendungskern.
+Dies liegt darin begründet, dass die Serviceschicht die Funktionalität des Anwendungskerns nutzt, um den Service bereitzustellen, und selbst keinerlei Geschäftslogik beinhaltet.
+
+[[fachkomponente-aufbau]]
+=== Aufbau einer Fachkomponente
+
+Eine Fachkomponente besteht aus einer *Service-Fassade* und einer *Exception-Fassade*.
+
+[[service-fassade]]
+==== Service-Fassade
+
+Die _Service-Fassade_ kümmert sich darum, eingehende Service-Aufrufe auf interne Aufrufe des Anwendungskerns abzubilden. 
+Transportobjekte der Serviceschicht werden dabei in Geschäftsobjekte des Anwendungskerns umgewandelt.
+Für die vom Anwendungskern gelieferten Ergebnisse ist eine Transformation in umgekehrter Richtung erforderlich. 
+
+Die Service-Fassade übernimmt darüber hinaus neben der Validierung der Aufrufparameter auch die Autorisierung des Aufrufs.
+Dazu nutzt sie den xref:isy-security:nutzungsvorgaben/master.adoc#autorisierung_service_schnittstelle[Baustein Security].
+Voraussetzung für die Autorisierung ist die Auswertung des mitgelieferten Access-Tokens.
+Über die Annotation `@Secured` wird die Berechtigung zum Zugriff auf die Methode anhand der Rechte des Aufrufers geprüft.
+Alternativ kann die Annotation `@Secured` auch an der Service-Fassade selbst verwendet werden, wenn alle ihre Methoden die gleichen Rechte erfordern.
+
+[[exception-fassade]]
+==== Exception-Fassade
+
+Die _Exception-Fassade_ ist verantwortlich für die Umwandlung der durch den Anwendungskern oder die Service-Fassade geworfenen Exceptions in Exceptions der Serviceschicht.
+Für das Ergebnis der Umwandlung ist sicherzustellen, dass entweder Exceptions ohne Stack-Traces oder ihnen entsprechende Transportobjekte zurückgegeben werden.
+Für die Behandlung und Umwandlung der aufgefangenen Exceptions sollten Log-Einträge erzeugt werden. 
+Dabei ist sicherzustellen, dass zum Zeitpunkt der Erstellung der Log-Einträge eine Korrelations-ID verfügbar ist.
+
+== Service-Framework
+
+// tag::service-framework-definition[]
+
+Das Service-Framework definiert die REST-Endpunkte des Backends und stellt diese zur Verfügung.
+Es wandelt die in der HTTP-Anfrage übertragenen Daten in Transportobjekte um und ruft die entsprechende Fachkomponente auf.
+Ebenso erledigt es die Umwandlung des Ergebnisses in eine HTTP-Antwort.
+Zur Definition der REST-Endpunkte wird Spring Web MVC bzw. Spring WebClient verwendet.
```

Comment:
Spring WebClient dient nicht zur *Definition* von REST-Endpunkten, sondern zu deren *Nutzung*  (und über die Nutzung (in Frontends oder anderen Backends) reden wir hier nicht.